modules/p2/ding_list/theme/theme.inc

```diff
@@ -269,21 +267,33 @@ function template_preprocess_ding_list_add_button(&$variables) {
   }
 
   if ($variables['type'] == 'ting_object') {
-    $new_list = array(
-      'custom' => array(
-        'title' => '<div class="list-title">' . t('New list') . '</div><div class="add-to-list-button button">' . t('Add to new list') . '</div>',
-        'href' => 'list/create',
-        'attributes' => array(
-          'class' => array(),
+    if (user_access('use personalisation')) {
+      $new_list = array(
+        'custom' => array(
+          'title' => '<div class="list-title">' . t('New list') . '</div><div class="add-to-list-button button">' . t('Add to new list') . '</div>',
+          'href' => 'list/create',
+          'attributes' => array(
+            'class' => array(),
+          ),
+          'query' => array('at' => $variables['type'], 'av' => $variables['value']),
+          'html' => TRUE,
         ),
-        'query' => array('at' => $variables['type'], 'av' => $variables['value']),
-        'html' => TRUE,
-      ),
-    );
+      );
+    }
+    elseif (!user_is_logged_in()) {
+      $new_list = array(
+        'custom' => array(
+          'title' => '<div class="list-title">' . t('Login before you can use lists') . '</div><div class="add-to-list-button button">' . t('Login') . '</div>',
+          'href' => '#',
+          'attributes' => array(
+            'class' => array('open-login'),
+          ),
+          'external' => TRUE,
+          'html' => TRUE,
 
-    if (!user_is_logged_in()) {
-      $new_list['custom']['attributes']['class'][] = 'use-ajax';
-      $new_list['custom']['href'] = 'ding-list/login-create';
+        ),
+      );
+      $force_multiple = TRUE;
```

Comment:
Why do we want to force multiple here in this case? Adding a comment explaining something non-obvious is nice.